src/primitives/transaction.h

```diff
@@ -200,6 +200,51 @@ class CTxOut
 extern boost::atomic<bool> flexTransActive;
 struct CMutableTransaction;
 
+template<typename Stream>
+void SerialiseScriptSig4(const std::vector<CTxIn> &inputs, Stream &s, int nType, int nVersion)
+{
+    for (auto in : inputs) {
+        bool first = true;
+        unsigned int i = 0; // use both iterator and int b/c the iterator doesn't do bounds-checks.
+        auto iter = in.scriptSig.begin();
+        while (i < in.scriptSig.size()) {
+            uint8_t k = *iter;
+            if ((k > 0 && k < 76) || (k >= 76 && k <= 78)) {
```

Comment:
I hear you. And check the PR request for a partial answer. I think this is hacky. But it is in line with the way that Bitcoin internals work today. There really is reinterpretation and re-serialisation of the tx all over the place.  I am not going to change it one step at a time.

The way to fix that properly is to have a different way of both designing and handling transactions inside of the app. And I think that has to be a separate project.